### DISCUSSION TITLE
Has the bootstrap lost momentum and is losing position to the projects like tailwind?

### DISCUSSION THREAD

**Discussion by ascensus-mdb** [twbs/bootstrap#31704] [General]
As a longtime and die-hard bootstrap user, I dare to ask an uncomfortable question - **is the bootstrap team observing the situation in their industry?**

Over the years, Bootstrap has been setting trends and standards on the frontend. And when you take into account the number of users and projects it is still the largest, but it seems like projects like tailwind are catching up very quickly.

I compare the dynamics of tailwind development - the number and speed of releasing new versions, the growing number of mentions on social media, growing traffic - and it is all impressive.

**And please don't take it personally because I'm only saying it out of concern** - as a bootstrap team don't you think bootstrap has lost momentum in development?

Few years of developing v4 or v5 is probably overkill. The same as many months of delays in releases despite the announcements of monthly releases.

All this makes people turn away from Bootstrap and go where they see greater dynamics and faster development - like tailwind for example.

I see and hear people saying "bootstrap is dead" more and more. But I don't want to believe it, and I can't accept it.

Therefore, I am asking you a question - do you also see these threats and problems, or is it just my imagination?

Sorry for any uncomfortable questions and thank you again for all your hard work.

**Comment by dawidadach** [comment]
I can't agree more :( 

This is an astonishing project which changed world of web development upside down!  I recently seen lot of discussion "Which framework to use" and it's very sad to see that people talk only about Tailwind and complain that Bootstrap 4 was "too heavy" or obsolete due to use of jQuery. 

Unfortunately currently there is a lot of hate on this amazing project on [Twitter ](https://twitter.com/search?q=bootstrap%20tailwind&src=typed_query) which is totally unfair. 

I've been playing with Bootstrap 5 since few months and I am sure that this gonna be a game changer! I am happy to see that even Alpha 1 is very stable so in my humble opinion Bootstrap should go as fast as possible to stable release before it's gonna be too late :( 

Please don't take it offensive , I really love Bootstrap 4 and even more Bootstrap 5 and want to give kudos to entire team making this fantastic project!

**Comment by mdo** [comment]
Agreed we have to move faster. We've taken a few hits lately with COVID and general burnout, so we've slowed down again. Hard to main two active versions at the same time, too. But! Hopefully these last couple weeks have shown momentum is slowly picking up again. We're super close to our second alpha—just 2 PRs left to review—and then we're shipping v4.5.3, and then v5 alpha 3 and v4.6.0.

We've pushed an update to our twbs/release repo to share some of these rough dates, but our goal is v5 stable end of year. Maybe even before then! Hoping to show folks we're not going anywhere :).

If you and others are looking to chip in, we could always use more help, especially on the JS side. Bug fixes, documentation updates, feature development, etc. Email me or ask around here for how best to help and we can ideally get some direction to folks before they get too far along. 

Thanks for the passion and appreciation! <3

**Comment by Hyp3rionM4x** [comment]
Hello all. So i know this discussion is only a few days/weeks old (?), but still wanted to take part of this discussion.

So your idea of BootStrap if being behind the curve at the moment, yes, could not agree more! I mean sure bootstrap 4 was a little heavy because of the use of things like jQuery and so on. HOWEVER! I still think, taking into consideration that they took all of this, and at the same time considered the old-browser-outdated-versions, being kept in the loop while they had to cover more ground of coding and such (not the best inside the subject, even tho i am an extreme fan of BS! but that is what i can only assume!).

Now i have been playing around for a few days now with bootstrap 5 too, and i am sure that it WILL be a gamechanger. Not only because they are dropping things like jQuery, but also because they are dropping the more outdated browser version compatibility and focus on the modern projects at hand - which i personally think is awesome btw!

There is a little way to go, sure! But keep in mind that we all have work to tend to one way or another to get income and we simply do not know what the story behind the bootstrap team is? Maybe there is focus at the project still but they are just not ready for it to show just yet what they are cooking behind the stage (if you know what i mean?) Here i am of course thinking about the bootstrap 5 that will probably bring some new (hopefully!!!!) components and such with it!

Myself for instance (just to mention something), have been struggling all the time with building accordian navigation menus and/or sidebars and of course search engines, which i think bootstrap should REALLY look into! I have already seen some few examples where another CSS Framework (Semantic UI), which have achieved all of this! And i really saw this added as a component/feature to bootstrap for my dynamic builds etc! I know this will not only help BootStrap grow ALOT! But also to bring more people back to BS again.

The search is done with JavaScript and the Accordian menu is done with HTML/CSS from what i can see. Same goes for stuff like SidePanels that would fill up the whole height of the users browser so you would have an easy way to build stuff like this.

I know this seems like a request but its not. I simply state that many ppl around (at least from what i read and see...) are looking for these things! Same goes for an easy layout of Calendars and the responsiveness behind it!

I can only hope the team will take my notes into feature-creation consideration.

And keep up the good work you are doing! :)

**Reply by mdo** [reply]
Definitely check out Alpha 2, just released today, and the Alpha 3 section in our blog post: https://blog.getbootstrap.com/2020/09/29/bootstrap-5-alpha-2/. Tons of things coming, including some improvements to what you're looking at :).

**Reply by Hyp3rionM4x** [reply]
Hmm all looks good - but i'm not sure i see what i was looking for. Try taking a good look at the ones i linked to. Accordion Menus - basically u would still have the options for a normal navbar BUT still have the option (seperated) to add another menu/accordion, and WITH that accordion, able to play around with stuff like placings/floats, etc so it would look like a side-panel etc. Much MUCH more smooth if you ask me AND i see a lot more people out there crying for the help, asking about side panels/menus/navbars especially! Some can achieve this in different ways with tutorials, sure, but if BootStrap could have this somehow included with samples so it would be like a "drag-&-drop" thing - are you crazy? MANY PEOPLE would look at you guys and think - wow, is this real? Finally no more looking.... in my case it's the same. I have been looking but many people usually one have 1 option and another have another option that you wanna combine but not possible because that specific tutorial have not thought of that and so you still need to use more brainy cells to think it out on your own :P 

Calendar, Accordion Menus, Tables, Searching Bars, etc. Those are the ones i'm looking for mostly - in my case that is.

For examples. Tables. I know that this is pretty hard doing with CSS (if not impossible - have not tested FLEX just yet fully!). But having a table row change color when hover is kinda easy. But having not only the SPECIFIC cell you hover change color AND! The (VERTICAL) part of the table columns change color too - see THAT! Would be AMAZING!

I know it probably would require some sort of JavaScripting - but it is small "unique" things like this, that makes BootStrap WORTH coming back to! Especially when its ideas like these that are being taken into consideration!

I hope you could use some of my words here ;) :D
And keep up the good work of course! <3

**Comment by patrickhlauke** [comment]
i would also comment that the aim is not to keep adding more and more new components. future refinements may, in fact, be simply better/more robust implementations of some of the same components, and even dropping certain components that are cluttering up the codebase but are now out of fashion, rarely used, whose use case is better solved by another component, etc.

**Reply by Hyp3rionM4x** [reply]
Hmm i get your idea too - but i would still like to see them be "available"? I mean you simply dont know if you come across say a custom that are looking for that specific design and then surprise surprise... bootstrap removed it? In my case i would love to see maybe a specific section - in this case "Components" improved and added MANY MORE features to it - and all the OTHER categories in bootstrap, keep them as simple as possible, sure.

I know it kinda sounds that im both for and off your suggestion, but i have simply had so many customers around me lately, where everyone of them are saying and asking for the very same thing - where i hoped BootStrap had it, but did not. So that is why i would love to keep it simple, but i would rather HAVE THE OPTION than not. If that makes sense? But good idea never the less to keep it simple xD

**Reply by ascensus-mdb** [reply]
I agree with @patrickhlauke . Bootstrap strength is being simple and lightweight. Do not go this messy way of adding new features and components. **Keep it simple**, because this is the reason people fall in love with bootstrap.

**Reply by Hyp3rionM4x** [reply]
> I agree with @patrickhlauke . Bootstrap strength is being simple and lightweight. Do not go this messy way of adding new features and components. **Keep it simple**, because this is the reason people fall in love with bootstrap.

i totally agree! But i would still love to see some sort of features on the side - you know - having the possibility to at least use them? So maybe split it up into 2 different libraries? css/js files i mean? so 1 css file that contains the simple version and then another with the features etc for those that have the interest in those? again - this is for those that wants the features (such as me)...

**Comment by ascensus-mdb** [comment]
Any predictions / dates about beta-2 and stable release?

**Reply by coliff** [reply]
The planned release schedule was updated last week here:
https://github.com/twbs/release#bootstrap-5
I think the best thing to do is be patient and keep your eye on the [Project board](https://github.com/twbs/bootstrap/projects) and calculate based on [remaining issues in the inbox](https://github.com/twbs/bootstrap/projects/33) for the forthcoming release.

**Comment by ascensus-mdb** [comment]
It seems that even this postponed schedule is not up to date anymore?

Can we expect a stable release in February as announced?

**Reply by patrickhlauke** [reply]
frankly...with the kind of shitshow that the last year and a bit has been on all of us, and considering that this is all done by volunteers in their spare time...may i suggest you just switch to tailwind at this point? it'd probably save you some hassle.

**Reply by mdo** [reply]
I'm sorry it's taking so long. People all over are struggling with life and work and so much more. Developing software isn't the only priority for us. We're trying to stay optimistic and productive, but personal life gets in the way and the things that can drop to the wayside do just that.

We're working on it as best we can while all staying sane.

**Reply by ascensus-mdb** [reply]
Of course, I didn't mean to offend anyone. If that's the case, I'm sorry.

I'm not going to switch to another framework. I consider Bootstrap to be number 1 in the world of web dev and want it to remain one.

That's the only reason I'm asking.

You've done a fantastic job creating Bootstrap. I feel very grateful for that and I would not like my words to demotivate anyone in these difficult times.

**Reply by mohe2015** [reply]
@ascensus-mdb You could try using the beta. I had good experience with it in the past. Of course this may not be useful for a big production project but I think it's fairly stable and bugfree already.
